frontend/modules/tree/tree.sagas.ts

```diff
@@ -49,7 +54,7 @@ export function* startListenOnSelections() {
 
 export function* getSelectedNodes() {
 	try {
-		yield call(delay, 0);
+		yield call(delay, 100);
```

Comment:
Honestly, it's a hack. As far as I know, it's because of the specific behavior of tree service. 
@dleszcz Could you tell more about it?

Reply:
@Charence It's temporary. We had to use some delay - similar to your fix in **groups.component** (old code with a comment below).

`this.$scope.$watchCollection(() => {
			return this.treeService.currentSelectedNodes;
		}, () => {
			this.$timeout(() => {
				/*
				 *	Temporary fix: This timeout is required because
				 * when currentSelectedNodes may be updated before unity
				 * (where groups get the object count) so without the delay
				 * it may have incorrect readings of how many objects are selected
				 * Proper fix would be to investigate this stupid control flow
				 * and make it work (and probably stop relaying on watches...)
				 */
				this.groupsService.updateSelectedObjectsLen().then(() => {
					this.groupsService.getSelectedObjects().then((currentHighlights) => {
						this.selectedNodes = currentHighlights || [];
						this.updateChangeStatus();
					});
				});
			});
		});` 

We will try to fix it and find a better solution when we will migrate TreeService.